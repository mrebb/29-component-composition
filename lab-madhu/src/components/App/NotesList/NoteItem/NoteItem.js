import React from 'react';
import NoteUpdateForm from './NoteUpdateForm/NoteUpdateForm';
export default class NoteItem extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      showComponent: false,
    };
  }

  onDoubleClick = e => {
    e.preventDefault();
    let showComponent = true;
    this.setState({ showComponent });
  };
  updateComponent = show => {
    let showComponent = show;
    this.setState({ showComponent });
  };
  render() {
    return (
      <React.Fragment>
        <li>
          <h2>{this.props.note.title}</h2>
          <p onDoubleClick={this.onDoubleClick}>{this.props.note.content}</p>
          {this.state.showComponent && (
            <NoteUpdateForm
              note={this.props.note}
              showComponentMethod={this.updateComponent}
              {...this.props}
            />
          )}
          <button onClick={() =>{ if (window.confirm('Are you sure you wish to delete this item?')) this.props.onRemove(this.props.note.id);}}>
            X
          </button>
        </li>
      </React.Fragment>
    );
  }
}
