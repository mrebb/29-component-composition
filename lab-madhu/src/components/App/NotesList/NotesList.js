import React from 'react';
import NoteItem from './NoteItem/NoteItem';

export default props => (
  <ul>
    {props.notes.length > 0 &&
      props.notes.map(note => (
        <NoteItem key={note.id} note={note} {...props} />
      ))}
  </ul>
);
