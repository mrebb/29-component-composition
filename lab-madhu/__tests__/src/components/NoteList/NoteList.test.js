import React from 'react';
import renderer from 'react-test-renderer';
import { shallow,mount } from 'enzyme';
import NotesList from '../../../../src/components/app/NotesList/NotesList';
import NoteItem from '../../../../src/components/App/NotesList/NoteItem/NoteItem';

describe('NoteList component', () => {
  it('is alive at application start', () => {
    let wrapper = mount(<NotesList notes={[]}/>);
    expect(wrapper.find(NoteItem)).toBeDefined();
  });
  xit('should render list of notes', done => {
    const onRemove = id => {
      expect(id).toBe('1234');
      done();
    };

    const notes = [];
    notes.push({
      id: '1234',
      title: 'foo',
      content: 'bar',
      editing: false,
      completed: true,
    });
    const wrapper = shallow(<NotesList notes={notes} onRemove={onRemove} />);
    expect(wrapper.find(NoteItem)).toExist();
    const noteItem = wrapper.find(NoteItem).first();
    expect (noteItem).toHaveProp('note');
    const noteItemId = noteItem.prop('note').id;

    const removeHandler = noteItem.prop('onRemove');

    removeHandler(noteItemId);
  });
});

describe('<NoteList/> (Snapshot Test)', () => {
  it('renders right', () => {
    const component = renderer.create(
      <NotesList notes={[]}/>
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();

  });
});
