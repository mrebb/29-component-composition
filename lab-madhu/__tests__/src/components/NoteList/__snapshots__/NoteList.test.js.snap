// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoteList/> (Snapshot Test) renders right 1`] = `<ul />`;

exports[`<NotesCreateForm/> (Snapshot Test) renders right 1`] = `<ul />`;
